distros/spring/src/main/java/com/connexta/transformation/service/TransformStatusController.java

```diff
@@ -0,0 +1,76 @@
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.service;
+
+import static com.connexta.transformation.service.PollResponseCreator.convertToPollResponse;
+
+import com.connexta.spring.error.DetailedResponseStatusException;
+import com.connexta.transformation.commons.api.TransformationManager;
+import com.connexta.transformation.commons.api.exceptions.TransformationException;
+import com.connexta.transformation.commons.api.exceptions.TransformationNotFoundException;
+import com.connexta.transformation.commons.api.status.Transformation;
+import com.connexta.transformation.rest.models.TransformationPollResponse;
+import com.connexta.transformation.rest.spring.TransformApiPoll;
+import com.connexta.transformation.service.Exceptions.UnknownStateException;
+import io.swagger.annotations.ApiParam;
+import java.net.URI;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+/** A rest controller for transformation status calls */
+@RestController
+public class TransformStatusController implements TransformApiPoll {
+
+  private final TransformationManager transformManager;
+
+  public TransformStatusController(TransformationManager transformManager) {
+    this.transformManager = transformManager;
+  }
+
+  /**
+   * For more documentation see {@link TransformApiPoll#poll(String)}. Retrieves a transformation
+   * with the given ID and creates a poll response from its status information.
+   *
+   * @param transformId The ID of the transformation to retrieve
+   * @param request The request
+   * @return A {@link TransformationPollResponse} containing the status information of the
+   *     transformation
+   * @throws TransformationNotFoundException if no transformation with the given ID could be found
+   * @throws TransformationException if an error occurs while retrieving the transformation
+   * @throws DetailedResponseStatusException if the state of the transformation could not be read
+   */
+  @Override
+  public ResponseEntity<TransformationPollResponse> poll(
+      @ApiParam(value = "The ID of the transform request. ", required = true)
+          @PathVariable("TransformId")
+          String transformId,
+      HttpServletRequest request)
+      throws TransformationException {
+    final Transformation transformation = transformManager.get(transformId);
+    final URI requestUri = URI.create(request.getRequestURI());
+    final TransformationPollResponse pollResponse;
+
+    try {
+      pollResponse = convertToPollResponse(transformation, requestUri);
```

Comment:
✏️ You could inline `URI.create(request.getRequestURI())` here instead of using the `requestUri` variable.